core/src/main/java/io/parsingdata/metal/expression/value/reference/Self.java

```diff
@@ -32,13 +32,24 @@
 public class Self implements ValueExpression {
 
     @Override
-    public ImmutableList<Optional<Value>> eval(final Environment environment, final Encoding encoding) {
-        return ImmutableList.create(Optional.of(environment.order.current()));
+    public ImmutableList<Optional<Value>> eval(final ParseGraph graph, final Encoding encoding) {
+        return ImmutableList.create(Optional.of(graph.current()));
     }
 
     @Override
     public String toString() {
         return getClass().getSimpleName();
     }
 
+    @Override
+    public boolean equals(Object obj) {
+        return obj != null
+            && getClass() == obj.getClass();
+    }
+
+    @Override
+    public int hashCode() {
+        return 2;
```

Comment:
I actually think `CurrentOffset` had 1, but since it was removed, now all that remains is 2.

(Remember, always two there are...)